tests/system/small/ml/test_llm.py

```diff
@@ -297,6 +311,20 @@ def test_embedding_generator_predict_series_success(
     assert len(value) == 768
 
 
+# @pytest.mark.flaky(retries=2)
+def test_embedding_generator_multi_cols_predict_success(
+    palm2_embedding_generator_model, llm_text_df
+):
+    llm_text_df["additional_col"] = 1
+    llm_text_df = llm_text_df.rename(columns={"prompt": "content"})
+    df = palm2_embedding_generator_model.predict(llm_text_df).to_pandas()
+    assert df.shape == (3, 5)
+    assert "text_embedding" in df.columns
+    series = df["text_embedding"]
+    value = series[0]
+    assert len(value) == 768
```

Comment:
nit: maybe we could coalesce line 323 - 325 into a single line?
assert len(df[..][0]) == 768

Reply:
sure, actually I'll rewrite the tests. Also some are already removed in a recent PR.

Reply:
done